credentials/apps/badges/models.py

```diff
@@ -265,6 +267,20 @@ def __str__(self):
 
     class Meta:
         verbose_name_plural = "Badge penalties"
+    
+    def apply_rules(self, data: dict) -> bool:
```

Comment:
@andrii-hantkovskyi can we get something like this:

```python
def apply_rules(self, data: dict) -> bool:
    return all(rule.apply(data) for rule in self.penaltydatarule_set.all())
```

Reply:
the same for requirements.

Reply:
let's also prettify
```
self.penaltydatarule_set.all() >> self.rules.all()
```

Reply:
done ^^